drivers/storage/portworx/component/autopilot.go

```diff
@@ -937,22 +920,41 @@ func (c *autopilot) isOCPUserWorkloadSupported() bool {
 	return *c.isUserWorkloadSupported
 }
 
-func (c *autopilot) refreshTokenSecret(secret *v1.Secret, clusterNamespace string, ownerRef *metav1.OwnerReference) error {
+// updateSecretTokenAndCert updates the secret with new token, token refresh timestamp and ca.crt
+func (c *autopilot) updateSecretTokenAndCert(secret *v1.Secret, clusterNamespace string, ownerRef *metav1.OwnerReference) error {
 
-	// update the token when it is half way to expiration
+	// update the token when it is half the way to expiration
 	newToken, err := generateAPSaToken(clusterNamespace, defaultAutoPilotSaTokenExpirationSeconds)
 	if err != nil {
 		return err
 	}
 
-	// update the secret with new token
-	secret.Data[core.ServiceAccountTokenKey] = []byte(newToken.Status.Token)
-	secret.Data[AutopilotSaTokenRefreshTimeKey] = []byte(time.Now().UTC().Add(time.Duration(*newToken.Spec.ExpirationSeconds/2) * time.Second).Format(time.RFC3339))
-	err = k8sutil.CreateOrUpdateSecret(c.k8sClient, secret, ownerRef)
+	// get the root ca.crt from the secret mounted inside the pod
+	// this is required while upgrading from OCP 4.15 to 4.16
+	// since the ca.crt is changed in 4.16 and not managed by openshift
+	rootCaCrt, err := getCertForSecret()
 	if err != nil {
 		return err
 	}
-	return nil
+
+	// update the secret with new token and secret
+	if secret.Data == nil {
+		secret.Data = make(map[string][]byte)
+	}
+	secret.Data[core.ServiceAccountTokenKey] = []byte(newToken.Status.Token)
+	secret.Data[core.ServiceAccountRootCAKey] = rootCaCrt
+	// set the token refresh time to half the way of token expiration
+	secret.Data[AutopilotSaTokenRefreshTimeKey] = []byte(time.Now().UTC().Add(time.Duration(*newToken.Spec.ExpirationSeconds/2) * time.Second).Format(time.RFC3339))
+	return k8sutil.CreateOrUpdateSecret(c.k8sClient, secret, ownerRef)
+}
+
+// get the root ca.crt from the secret mounted inside the pod
+func getCertForSecret() ([]byte, error) {
+	rootCaCrt, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/ca.crt")
+	if err != nil && !os.IsNotExist(err) {
+		return nil, fmt.Errorf("error reading k8s cluster certificate located inside the pod at /var/run/secrets/kubernetes.io/serviceaccount/ca.crt: %w", err)
```

Comment:
nit: since the `/var/run/secrets/.../ca.crt` path is mentioned multiple times, we should put it in a const